packages/headless/src/controllers/product-listing/results-per-page/headless-product-listing-results-per-page.ts

```diff
@@ -0,0 +1,65 @@
+import {configuration, pagination} from '../../../app/reducers';
+import {logPagerResize} from '../../../features/pagination/pagination-analytics-actions';
+import {fetchProductListing} from '../../../features/product-listing/product-listing-actions';
+import {ProductListingEngine} from '../../../product-listing.index';
+import {
+  ConfigurationSection,
+  PaginationSection,
+} from '../../../state/state-sections';
+import {loadReducerError} from '../../../utils/errors';
+import {
+  buildCoreResultsPerPage,
+  ResultsPerPage,
+  ResultsPerPageInitialState,
+  ResultsPerPageProps,
+  ResultsPerPageState,
+} from '../../core/results-per-page/headless-core-results-per-page';
+
+export {
+  ResultsPerPage,
+  ResultsPerPageProps,
+  ResultsPerPageInitialState,
+  ResultsPerPageState,
+};
+
+/**
+ * Creates a `ResultsPerPage` controller instance.
+ *
+ * @param engine - The headless engine.
+ * @param props - The configurable `ResultsPerPage` properties.
+ * @returns A `ResultsPerPage` controller instance.
+ */
+export function buildResultsPerPage(
+  engine: ProductListingEngine,
+  props: ResultsPerPageProps = {}
+): ResultsPerPage {
```

Comment:
Ah, I missed that in other PRs. Perhaps we should ensure that in all cases where we have a `core`, a `product-listing` and headless(?) component we should ensure we have 3 separate interfaces (where the `product-listing` and headless interface extend the `core` interface).

I'm worried that if/when functionalities diverge between the two, the changes are going to be breaking and developers using commerce are going to be forced to change the interfaces they use.

Reply:
Hmm, I think we could create the core interface when the behavior diverges. I'm assuming that the product-listing controller will still use a `ResultsPerPage` instead of prefixing with `ProductListing` to avoid a breaking change.

Ofc, there would be multiple interfaces with the same name, but I think that is ok, aligned with the name of the controller function. There should be no collisions since the interfaces would be exposed from independent sub-packages. The alternative would be to prefix, but I think that makes the names too long, and duplicates the name of the subpackage already specified in the import path. e.g.

```
import {buildResultsPerPage, ResultsPerPage} from "@coveo/headless/product-listing";

//vs

import {
  buildProductListingResultsPerPage,
  ProductListingResultsPerPage
} from "@coveo/headless/product-listing";
```

Reply:
Unless I'm mistaken, I thought the point was that the functionalities _not_ diverge so that whether you're building a classic Search interface or a Product listing page, the experience be the same but with different imports?